Input: synaptics-rmi4 - Supports to query DPM value.

myangsynapticscom opened the pull request:
RMI4 F12 will support to query DPM value on Touchpad. When TP firmware doesn't support to report logical and physical value within the Touchpad's HID report.
We can directly query the DPM value through RMI.

myangsynaptics committed:
Input: synaptics-rmi4 - Supports to query DPM value.

RMI4 F12 will support to query DPM value on Touchpad.
When TP firmware doesn't support to report logical and
physical value within the Touchpad's HID report.
We can directly query the DPM value through RMI.

Signed-off-by: Marge Yang <marge.yang@tw.synaptics.com>
Signed-off-by: Vincent Huang <Vincent.Huang@tw.synaptics.com>

vhuag commented on drivers/input/rmi4/rmi_f12.c:
```diff
@@ -24,6 +24,7 @@ enum rmi_f12_object_type {
 };
 
 #define F12_DATA1_BYTES_PER_OBJ			8
+#define RMI_QUERRY_DPM_IN_PRESENSE_BIT 29
```
typo here.

vhuag commented on drivers/input/rmi4/rmi_f12.c:
```diff
@@ -329,6 +330,9 @@ static int rmi_f12_probe(struct rmi_function *fn)
 	u16 data_offset = 0;
 	int mask_size;
 
+	u16 query_dpm_addr = fn->fd.query_base_addr;
```
why do we init as query base?

myangsynaptics committed:
Input: synaptics-rmi4 - Supports to query DPM value.

RMI4 F12 will support to query DPM value on Touchpad.
When TP firmware doesn't support to report logical and
physical value within the Touchpad's HID report.
We can directly query the DPM value through RMI.

Signed-off-by: Marge Yang <marge.yang@tw.synaptics.com>
Signed-off-by: Vincent Huang <Vincent.Huang@tw.synaptics.com>

vhuag commented on drivers/input/rmi4/rmi_f12.c:
```diff
@@ -428,6 +447,10 @@ static int rmi_f12_probe(struct rmi_function *fn)
 		return ret;
 
+	if (support_dpm_query) {
+		sensor->x_mm = sensor->max_x / dpm_resolution;
```
can we move to line 430?
